File: Python/OCR.py
import requests
import json
from qiniu import QiniuMacAuth

access_key = '3gFTmZm'
secret_key = 'aF3_h9AwDE'


#生成鉴权QiniuToken
auth = QiniuMacAuth(access_key,secret_key)
body = json.dumps(
{
    "data": {
        "uri": "http://"
    }
}

)

url = "http://ai.qiniuapi.com/v1/ocr/idcard"
token = "Qiniu " + auth.token_of_request(method="POST",url = url,body=body,content_type="application/json")

header = {
    "Authorization":token,
    "Host":"ai.qiniuapi.com",
    "Content-Type":"application/json"
}

response = requests.post(url,headers = header,data=body)
print(response.status_code,response.text)

File: Java/OCR.java
 // OCR身份证识别接口
    public static void main(String[] args) {

       // 构造一个带指定Zone对象的配置类
        Configuration cfg = new Configuration(Zone.zone0());
        String accessKey = "RrnoWYRZS5qrZBYqkQ2RNAcCTwtcqCiBRFXFlWej";
        String secretKey = "eFZjKLqv2z1JEDwFqUjUi7BhFma0Y4S9836asMqz";
        //默认不指定key的情况下，以文件内容的hash值作为文件名
        String url = "http://ai.qiniuapi.com/v1/ocr/idcard";
        // 构造post请求body
        Gson gson = new Gson();
        Map<String, String> m = new HashMap();
        // 这里放在空间内身份证的链接
        m.put("uri", "http://pd4ivutut.bkt.clouddn.com/0f7f45b8-c1c2-4742-a20c-469a38cd4408.jpg");
        Map<String, Object> m1 = new HashMap();
        m1.put("data", m);
        String paraR = gson.toJson(m1);
        byte[] bodyByte = paraR.getBytes();
        Auth auth = Auth.create(accessKey, secretKey);
        // 生成token
        String token = "Qiniu " + auth.signRequestV2(url, "POST", bodyByte, "application/json");
        BucketManager bucketManager = new BucketManager(auth, cfg);

        Client client = new Client();
        // 请求头构建
        StringMap headers = new StringMap();
        headers.put("Authorization", token);
        headers.put("Host", "ai.qiniuapi.com");
        headers.put("Content-Type", "application/json");
        try {
            com.qiniu.http.Response resp = client.post(url, bodyByte, headers, Client.JsonMime);
            System.out.println(resp);
            System.out.println(resp.bodyString());
//            return resp.bodyString();
        } catch (Exception e) {
            System.out.println(e);
        }

    }